lib/rules/no-reserved-component-names.js

```diff
@@ -0,0 +1,113 @@
+/**
+ * @fileoverview disallow the use of reserved names in component definitions
+ * @author Jake Hassel <https://github.com/shadskii>
+ */
+'use strict'
+
+const utils = require('../utils')
+const casing = require('../utils/casing')
+
+const htmlElements = require('../utils/html-elements.json')
+const svgElements = require('../utils/svg-elements.json')
+
+const kebabCaseElements = [
+  'annotation-xml',
+  'color-profile',
+  'font-face',
+  'font-face-src',
+  'font-face-uri',
+  'font-face-format',
+  'font-face-name',
+  'missing-glyph'
+]
+
+const isLowercase = (word) => (/[a-z]/.test(word))
+const capitalizeFirstLetter = (word) => word[0].toUpperCase() + word.substring(1, word.length)
+
+const RESERVED_NAMES = new Set(
+  [
+    ...kebabCaseElements,
+    ...kebabCaseElements.map(casing.pascalCase),
+    ...htmlElements,
+    ...htmlElements.map(capitalizeFirstLetter),
+    ...svgElements,
+    ...svgElements.filter(isLowercase).map(capitalizeFirstLetter)
+  ])
+
+// ------------------------------------------------------------------------------
+// Rule Definition
+// ------------------------------------------------------------------------------
+
+module.exports = {
+  meta: {
+    type: 'suggestion',
+    docs: {
+      description: 'disallow the use of reserved names in component definitions',
+      category: 'essential',
```

Comment:
Please set undefined.

```suggestion
      category: undefined, // 'essential'
```

Categories are set at the timing of the major version up.

Reply:
My mistake, I have corrected this with your suggestion!